workspaces/libnpmaccess/lib/index.js

```diff
@@ -1,186 +1,140 @@
-      ...opts,
-      method: 'DELETE',
-      body: { package: spec.name },
-      scope,
-      spec,
-      ignoreBody: true,
-    })
-      .then(() => true)
-  })
+const getCollaborators = async (pkg, opts) => {
+  const spec = npar(pkg)
+  const uri = `/-/package/${spec.escapedName}/collaborators`
+  return npmFetch.json(uri, opts)
 }
 
-cmd.lsPackages = (entity, opts) => {
-  return cmd.lsPackages.stream(entity, opts)
-    .collect()
-    .then(data => {
-      return data.reduce((acc, [key, val]) => {
-        if (!acc) {
-          acc = {}
-        }
-        acc[key] = val
-        return acc
-      }, null)
-    })
+const getVisibility = async (pkg, opts) => {
+  const spec = npar(pkg)
+  const uri = `/-/package/${spec.escapedName}/visibility`
+  return npmFetch.json(uri, opts)
 }
 
-cmd.lsPackages.stream = (entity, opts = {}) => {
-  validate('SO|SZ', [entity, opts])
-  const { scope, team } = splitEntity(entity)
-  let uri
-  if (team) {
-    uri = `/-/team/${eu(scope)}/${eu(team)}/package`
-  } else {
-    uri = `/-/org/${eu(scope)}/package`
-  }
-  const nextOpts = {
+const setAccess = async (pkg, access, opts) => {
+  const spec = npar(pkg)
+  const uri = `/-/package/${spec.escapedName}/access`
+  await npmFetch(uri, {
     ...opts,
-    query: { format: 'cli' },
-    mapJSON,
-  }
-  const ret = new Minipass({ objectMode: true })
-  npmFetch.json.stream(uri, '*', nextOpts)
-    .on('error', err => {
-      if (err.code === 'E404' && !team) {
-        uri = `/-/user/${eu(scope)}/package`
-        npmFetch.json.stream(uri, '*', nextOpts)
-          .on('error', streamErr => ret.emit('error', streamErr))
-          .pipe(ret)
-      } else {
-        ret.emit('error', err)
-      }
-    })
-    .pipe(ret)
-  return ret
-}
-
-cmd.lsCollaborators = (spec, user, opts) => {
-  return Promise.resolve().then(() => {
-    return cmd.lsCollaborators.stream(spec, user, opts)
-      .collect()
-      .then(data => {
-        return data.reduce((acc, [key, val]) => {
-          if (!acc) {
-            acc = {}
-          }
-          acc[key] = val
-          return acc
-        }, null)
-      })
+    method: 'POST',
+    body: { access },
+    spec,
+    ignoreBody: true,
   })
+  return true
 }
 
-cmd.lsCollaborators.stream = (spec, user, opts) => {
-  if (typeof user === 'object' && !opts) {
-    opts = user
-    user = undefined
-  } else if (!opts) {
-    opts = {}
+const setMfa = async (pkg, level, opts) => {
+  const spec = npar(pkg)
+  const body = {}
+  switch (level) {
+    case 'none':
+      body.publish_requires_tfa = false
+      break
+    case 'publish':
+      // tfa is required, automation tokens can not override tfa
+      body.publish_requires_tfa = true
+      body.automation_token_overrides_tfa = false
+      break
+    case 'automation':
+      // tfa is required, automation tokens can override tfa
+      body.publish_requires_tfa = true
+      body.automation_token_overrides_tfa = true
+      break
```

Comment:
Can we create automation tokens via the registry APIs? I thought that wasn't actually possible from previous conversations

Reply:
It does work, I've done it before. We just need to add an additional argument. I had it working with a monkey patched version of the CLI previously.

Reply:
These two `body` arguments are the arguments. Everything in this PR works from my testing locally.

Reply:
=`readonly`, `automation`, and `cidr_whitelist` are supported on the registry right now. `token_name` is not.